src/java/htsjdk/samtools/cram/encoding/huffman/codec/HuffmanParamsCalculator.java

```diff
@@ -0,0 +1,153 @@
+package htsjdk.samtools.cram.encoding.huffman.codec;
+
+import htsjdk.samtools.cram.common.MutableInt;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanCode;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanTree;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * The purpose is to calculate Huffman encoding parameters.
+ * Critical: ALL values must be shown to the calculator before calling the calculate() method.
+ */
+public class HuffmanParamsCalculator {
+    private final HashMap<Integer, MutableInt> countMap = new HashMap<Integer, MutableInt>();
```

Comment:
Thank you. The difference for 10^2 elements will indeed be negligible. 

There is, however, one more advantage of specialized collections: you don't need `MutableInt`, because there won't be redundant allocations.